### PR TITLE
API definition

### PR DESCRIPTION
**Pull request by feeds**

### CODE REVIEW

File: api/server.py [hunk 1]
```diff
@@ -1,12 +1,62 @@
...
 
 app = Flask(__name__)
+CORS(app, resources={r"/*": {"origins": "http://localhost:{}".format(REACT_PORT)}})
 
 
 def run(port, hostname, debug_mode):
     app.run(host=hostname, port=port, debug=debug_mode)
 
 
-@app.route("/")
```
Comment (GittiHab) [inline api/server.py]: Wouldn't it be nice to have a root route?

File: api/server.py [hunk 2]
```diff
@@ -1,12 +1,62 @@
...
-def hello_world():
+# TODO: If meta-paths for A and B will be written in Java, they will need this information in Java
```
Comment (GittiHab) [inline api/server.py]: What do you mean by this?

File: api/server.py [hunk 3]
```diff
@@ -1,12 +1,62 @@
...
+@app.route("/node-sets", methods=["POST"])
+def receive_node_sets():
```
Comment (GittiHab) [inline api/server.py]: Is this also for active learning?

File: api/server.py [hunk 4]
```diff
@@ -1,12 +1,62 @@
...
+    # TODO: Check if necessary information is in request object
+    if not request.json:
+        abort(400)
+
+
+@app.route("/node-sets", methods=["GET"])
+def send_node_sets():
```
Comment (GittiHab) [inline api/server.py]: I don't like the naming yet. Send would implicate that this function sends something somewhere else async. But actually someone is pulling information, right? How about actually naming it `get_node_sets`?

File: api/server.py (hunk 3 above)
Comment (GittiHab) [inline api/server.py]: Also he name could be `add_node_sets` (see the next comment).

File: api/server.py [hunk 5]
```diff
@@ -1,12 +1,62 @@
...
+    # TODO: Call fitting method in active_learning
+    return jsonify("Hello world")
+
+
+# TODO: If meta-paths for A and B will be written in Java, they will need this information in Java
```
Comment (GittiHab) [inline api/server.py]: Same as before...

File: api/server.py [hunk 6]
```diff
@@ -1,12 +1,62 @@
...
+@app.route("/types", methods=["POST"])
+def receive_edge_node_types():
```
Comment (GittiHab) [inline api/server.py]: Same as before...

File: api/server.py (hunk 3 above)
Comment (feeds) [reply api/server.py]: no, this is the input on page one

**Pull request merged**